build.sh

```diff
@@ -423,10 +423,13 @@ case "$COMMAND" in
         ln -s "RealmSwift-swift$version" RealmSwift
         git update-index --assume-unchanged RealmSwift || true
 
-        SWIFT_VERSION_FILE="RealmSwift/SwiftVersion.swift"
-        CONTENTS="let swiftLanguageVersion = \"$version\""
-        if [ ! -f "$SWIFT_VERSION_FILE" ] || ! grep -q "$CONTENTS" "$SWIFT_VERSION_FILE"; then
-            echo "$CONTENTS" > "$SWIFT_VERSION_FILE"
+        # Only write SwiftVersion.swift if RealmSwift supports the given version of Swift.
+        if [[ -e "RealmSwift-swift$version" ]]; then
```

Comment:
Why not move this check higher in `set-swift-version`, failing as early as possible when building with an unsupported version of Swift, avoiding the extra script that was added in your later commit?